src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -214,7 +215,7 @@ public Set<String> getAgencyIds(boolean allowInactive) {
      * @throws AccessDeniedException   if current user does not have access to this agency and accessDeniedError is true.
      */
     public void verifyAgencyAccess(final String agencyId, boolean accessDeniedError, boolean allowInactive) {
-        Objects.requireNonNull(agencyId, "agencyId is a required parameter");
+        checkAgencyExists(agencyId, allowInactive ? ALL: ACTIVE_ONLY);
```

Comment:
Not sure about this - does it matter if a 403 is returned when the agency doesnt exist? Just thinking about the extra db calls 🤷

Reply:
As discussed - move checks into non-happy path